src/common/hooks/useArrayStateManager.js

```diff
@@ -45,6 +50,8 @@ function useArrayStateManager({ maxItems }) {
     handleSelectArray,
     handleRemoveItem: (item) =>
       setSelectedItems(removeItemFromArray(selectedItems, item)),
+    handleRemoveAllItems: (items) =>
```

Comment:
Minor: let's call these new things removeItems rather than removeAllItems. "Remove all" makes me think we're going to wipe out the entire array not the the items that got passed in